controllers/status.go

```diff
@@ -69,8 +72,34 @@ func (r *EtcdadmClusterReconciler) updateStatus(ctx context.Context, ec *etcdv1.
 
 	// etcd ready when all machines have address set
 	ec.Status.Ready = true
-	ec.Status.Endpoints = strings.Join(endpoints, ",")
 	conditions.MarkTrue(ec, etcdv1.EtcdEndpointsAvailable)
+
+	sort.Strings(endpoints)
+	ec.Status.Endpoints = strings.Join(endpoints, ",")
+
+	log.Info("Comparing current and previous endpoints")
+	// Saving the current endpoints as an annotation, so that we can check if endpoints have changed. This avoids unnecessary client calls
+	// to get and update the Secret containing the endpoints
```

Comment:
nit: No big deal but my vote would be to avoid this.

It only saves us one call (get secret) and only in the best worst scenario (no changes). In the best case scenario, it doesn't save us anything.

The main reason is this seems like early optimization that adds a bit of complexity (now we need to maintain two fields in sync) but it's solving a problem that we don't necessarily know if it exists.

Also, side comment. If we just want to compare with last value, why don't we just compare against the status itself? It seems like the status is also updated in this method, which makes me think that we can just compare `strings.Join(endpoints, ",")` against `ec.Status.Endpoints`.

Reply:
oh right I can just compare them